apps/web/src/components/Ecosystem/Content.tsx

```diff
@@ -42,51 +66,90 @@ const decoratedEcosystemApps: EcosystemApp[] = orderedEcosystemAppsAsc().map((d)
   searchName: d.name.toLowerCase(),
 }));
 
+const updateUrlParams = (params: { categories?: string[]; subcategories?: string[] }) => {
+  console.log('categories:', params.categories);
+
+  console.log('subcategories:', params.subcategories);
+  const searchParams = new URLSearchParams(window.location.search);
+
+  if (params.categories?.length) {
+    searchParams.set('category', params.categories.join(','));
+  } else {
+    searchParams.delete('category');
+  }
+
+  if (params.subcategories?.length) {
+    searchParams.set('subcategory', params.subcategories.join(','));
+  } else {
+    searchParams.delete('subcategory');
+  }
+
+  window.history.pushState(
+    {},
+    '',
+    `${window.location.pathname}${searchParams.toString() ? '?' + searchParams.toString() : ''}`,
+  );
+};
+
 export default function Content() {
-  const [selectedTags, setSelectedTags] = useState<string[]>(['all']);
-  const [search, setSearch] = useState<string>('');
+  const [search, setSearch] = useState('');
   const [showCount, setShowCount] = useState<number>(16);
+  const searchParams = useSearchParams();
+  const [selectedSubcategories, setSelectedSubcategories] = useState<string[]>(() => {
+    const subcategories = searchParams?.get('subcategory');
+    return subcategories ? subcategories.split(',') : [];
+  });
 
-  const selectTag = (tag: string): void => {
-    setSelectedTags((prevTags) => {
-      if (tag === 'all') {
-        return ['all'];
-      }
-      const newTags = prevTags.includes(tag)
-        ? prevTags.filter((t) => t !== tag)
-        : [...prevTags.filter((t) => t !== 'all'), tag];
-      return newTags.length === 0 ? ['all'] : newTags;
-    });
-  };
+  // If a subcategory is selected, the category is selected automatically
+  const selectedCategories = useMemo(() => {
+    return [
+      ...new Set(
+        selectedSubcategories.map(
+          (subcategory) =>
+            Object.keys(config).find((category) => config[category].includes(subcategory)) ?? 'all',
+        ),
+      ),
+    ];
+  }, [selectedSubcategories]);
```

Comment:
nit, but stylistically we're moving towards 
```
() => ([
...
])
```

instead of brackets with return statements